docs/api/hardhat/hardhat-zksync-upgradable.md

```diff
@@ -0,0 +1,433 @@
+
+To deploy a simple upgradable contract on the zkSync local setup, first create a test wallet and add it to the new Deployer.
+
+```typescript
+
+  const testMnemonic = 'stuff slice staff easily soup parent arm payment cotton trade scatter struggle';
+  const zkWallet = Wallet.fromMnemonic(testMnemonic, "m/44'/60'/0'/0/0");
+
+  const deployer = new Deployer(hre, zkWallet);
+```
+
+After that, simply load the Box artifact and call the `deployProxy` method from the `zkUpgrades` hre property.
+
+```typescript
+  const contract = await deployer.loadArtifact(contractName);
+  await hre.zkUpgrades.deployProxy(deployer.zkWallet, contract, [42], { initializer: 'store' });
+```
+
+And that is it! <br/>
+The `deployProxy` method will deploy your implementation contract on the zkSync network, deploy the proxy admin contract, and finally, deploy the transparent proxy.
+
+Here is one example of the whole deploy proxy script:
+
+```typescript
+
+import { Deployer } from '@matterlabs/hardhat-zksync-deploy';
+import { Wallet } from 'zksync-web3';
+
+const hre = require('hardhat');
+
+async function main() {
+    const contractName = 'Box';
+    console.log('Deploying ' + contractName + '...');
+
+    const testMnemonic = 'stuff slice staff easily soup parent arm payment cotton trade scatter struggle';
+    const zkWallet = Wallet.fromMnemonic(testMnemonic, "m/44'/60'/0'/0/0");
+
+    const deployer = new Deployer(hre, zkWallet);
+
+    const contract = await deployer.loadArtifact(contractName);
+    const box = await hre.zkUpgrades.deployProxy(deployer.zkWallet, contract, [42], { initializer: 'store' });
+
+    await box.deployed();
+    console.log(contractName + ' deployed to:', box.address);
+
+    box.connect(zkWallet);
+    const value = await box.retrieve();
+    console.log('Box value is: ', value.toNumber());
+}
+
+main();
+
+
+```
+
+> **_IMPORTANT NOTE:_** `deployProxy` method (and other deploy/upgrade methods from the zkUpgrades) needs the know which wallet to use to deploy smart contracts. For this reason, the wallet needs to have a configured provider that connects it to the specific zkSync network. This provider is configured in the hardhat config file, by stating the RPC url of the network to connect to.
+
+```typescript
+
+defaultNetwork: 'zkSyncNetwork',
+    networks: {
+        goerli: {
+            zksync: false,
+            url: 'http://localhost:3050',
+        },
+        zkSyncNetwork: {
+            zksync: true,
+            ethNetwork: 'goerli',
+            url: 'http://localhost:8545',
+        },
+    },
+
+```
+
+Since the provider instatiation is already handled in the creation of the Deployer class based on your hardhat configuration, we can simply pass the `deployer.zkWallet` and be sure that the correct provider is already set.
```

Comment:
```suggestion
Since the provider was instantiated on creating the `Deployer` class based on your Hardhat configuration, we only have to pass the `deployer.zkWallet` to be sure that the correct provider is already set.
```